datafusion/physical-expr/src/physical_expr.rs

```diff
@@ -136,6 +138,67 @@ impl PhysicalExprStats for BasicExpressionStats {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct EquivalenceProperties {
```

Comment:
I wonder if `EquivalenceClass` is a more specific name?

Then you could make

```rust
struct EquivalenceProperties {
  classes: Vec<EquivalentClass>
}
```

And move functions like `truncate_equivalence_properties_not_in_schema` on to 

```rust
impl EquivalenceProperties {
  fn truncate_equivalence_properties_not_in_schema(&self, ..)
}
```

I don't think it is required, but it might keep the code easier to reason about / keep it behind an abstraction

Reply:
Having `EquivalenceProperties` would also provide a single location to add docstrings explaining the structures, and their assumptions and what they are good for

Reply:
Sure.

Reply:
Hi @alamb,  is there a kind of linked hash map in Rust? Maybe better to leverage a similar data structure here.

Reply:
The preferred data structure is LinkedHashSet.